### PR TITLE
Check node argument is defined before use

### PR DESCRIPTION
**Pull request by G-Rath**
This fixes errors like this one in `no-empty-title`:

```
TypeError: Cannot read property 'type' of undefined
Occurred while linting C:\Users\G-Rath\workspace\projects-personal\repo-event-notifier-lambda\test\src\utils\isRequestFromSlack.spec.ts:27

TypeError: Cannot read property 'type' of undefined
Occurred while linting C:\Users\G-Rath\workspace\projects-personal\repo-event-notifier-lambda\test\src\utils\isRequestFromSlack.spec.ts:27
    at isString (C:\Users\G-Rath\workspace\projects-personal\repo-event-notifier-lambda\node_modules\eslint-plugin-jest\lib\rules\util.js:86:31)
    at CallExpression (C:\Users\G-Rath\workspace\projects-personal\repo-event-notifier-lambda\node_modules\eslint-plugin-jest\lib\rules\no-empty-title.js:46:14)
    at listeners.(anonymous function).forEach.listener (C:\Users\G-Rath\workspace\projects-personal\repo-event-notifier-lambda\node_modules\eslint\lib\util\safe-emitter.js:45:58)
    at Array.forEach (<anonymous>)
    at Object.emit (C:\Users\G-Rath\workspace\projects-personal\repo-event-notifier-lambda\node_modules\eslint\lib\util\safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (C:\Users\G-Rath\workspace\projects-personal\repo-event-notifier-lambda\node_modules\eslint\lib\util\node-event-generator.js:251:26)
    at NodeEventGenerator.applySelectors (C:\Users\G-Rath\workspace\projects-personal\repo-event-notifier-lambda\node_modules\eslint\lib\util\node-event-generator.js:280:22)
    at NodeEventGenerator.enterNode (C:\Users\G-Rath\workspace\projects-personal\repo-event-notifier-lambda\node_modules\eslint\lib\util\node-event-generator.js:294:14)
    at CodePathAnalyzer.enterNode (C:\Users\G-Rath\workspace\projects-personal\repo-event-notifier-lambda\node_modules\eslint\lib\code-path-analysis\code-path-analyzer.js:632:23)
    at nodeQueue.forEach.traversalInfo (C:\Users\G-Rath\workspace\projects-personal\repo-event-notifier-lambda\node_modules\eslint\lib\linter.js:752:32)
Process finished with exit code -1
```

This was caused by a lack of definition guard:

https://github.com/jest-community/eslint-plugin-jest/blob/d7a9532ba978c2d5b5d94e728bcbc11cfe158429/src/rules/no-empty-title.js#L33-L36

I figured there is no harm in adding `node &&` to all the util methods. 
Ironically this (hopefully) would be caught when converting to typescript 😉

### CODE REVIEW

File: src/rules/util.js [hunk 1]
```diff
@@ -125,18 +125,22 @@ const isTestCase = node =>
...
+  describeAliases[getNodeName(node.callee)];
 
 const isFunction = node =>
-  node.type === 'FunctionExpression' || node.type === 'ArrowFunctionExpression';
+  (node && node.type === 'FunctionExpression') ||
+  node.type === 'ArrowFunctionExpression';
 
 const isString = node =>
-  (node.type === 'Literal' && typeof node.value === 'string') ||
+  (node && (node.type === 'Literal' && typeof node.value === 'string')) ||
```
Comment (G-Rath) [inline src/rules/util.js]: I didn't notice the prettier put `node` in this condition, instead of outside 😬

**Comment by G-Rath** [comment]
Random question: for some reason there are a couple of seemingly stray strings in the output on travis:

```
...
 PASS   lint  src/rules/__tests__/require-tothrow-message.test.js
 PASS   lint  src/rules/prefer-to-have-length.js
 PASS   lint  src/rules/__tests__/no-hooks.test.js
 PASS 27m  lint  src/rules/__tests__/prefer-to-be-null.test.js
 PASS   lint  src/rules/prefer-inline-snapshots.js
 PASS   lint  src/rules/no-hooks.js
 PASS   lint  src/rules/prefer-to-be-undefined.js
 PASS   lint  src/rules/__tests__/no-jest-import.test.js
 PASS   lint  src/rules/prefer-to-be-null.js
 PASS   lint  src/processors/__tests__/snapshot-processor.test.js
 PASS   lint  src/rules/__tests__/prefer-called-with.js
 PASS   lint  src/rules/no-test-prefixes.js
 PASS   lint  src/rules/no-empty-title.js
 PASS   lint  src/rules/no-mocks-import.js
 PASS   lint  src/rules/no-truthy-falsy.js
 PASS   lint  src/rules/no-test-return-statement.js
 PASS   lint  src/rules/__tests__/no-test-return-statement.test.js
 PASS   lint [27m src/__tests__/rules.test.js
 PASS   lint  src/rules/__tests__/prefer-to-have-length.test.js
 PASS   lint  src/rules/require-tothrow-message.js
 PASS   lint  src/rules/prefer-called-with.js
.. and so on
``` 
Two `27m`s; one on the line for `prefer-to-be-null.test.js` & another on the line for `rules.test.js`

```
File                          |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
------------------------------|----------|----------|----------|----------|-------------------|
All files                     |      100 |      100 |      100 |      100 |                   |
 src                          |      100 |      100 |      100 |      100 |                   |
  index.js                    |      100 |      100 |      100 |      100 |                   |
 src/processors               |      100 |      100 |      100 |      100 |                   |
  snapshot-processor.js       |      100 |      100 |      100 |      100 |                   |
 src/rules                    |      100 |      100 |      100 |      100 |                   |
  consistent-test-it.js       |      100 |      100m |      100 |      100 |                   |
  expect-expect.js            |      100 |      100 |      100 |      100 |                   |
  lowercase-name.js           |      100 |      100 |      100 |      100 |                   |
  no-alias-methods.js         |      100 |      100 |      100 |      100 |                   |
... and so on
```
A stray `m` for `consistent-test-it.js`.

Anyone know what's up w/ that, or is it just random out-of-sync console output?

**Comment by SimenB** [comment]
Could you add a test that fails without this change?

**Comment by G-Rath** [comment]
I was a bit worried you'd ask me that, as there are currently no tests for utils.

Would it be enough to add a test for `no-empty-title`? That'd only cover `isString`, and not sure if there are other rules that use the other methods in a manner that I can make fail.

Otherwise, I can just write a test where I pass `undefined` to each method, if that's acceptable :joy:

**Comment by SimenB** [comment]
A test for `no-empty-title` is perfect 🙂

**Comment by G-Rath** [comment]
Done 👍
